snap/local/README.md

```diff
@@ -0,0 +1,411 @@
+
+If build success, you can find `deviceupdate-agent_#.#_amd64.snap` at the project root directory.
+
+### How To View the DU Snap Package Content
+
+You can use `unsquashfs` command to extract `.snap` file.
+
+```shell
+# Run unsquashfs <snap file name>
+# e.g., unsquashfs deviceupdate-agent_0.1_amd64.snap
+rm -fr squashfs-root
+
+unsquashfs deviceupdate-agent_0.1_amd64.snap
+
+# The content will be extracted to 'squashfs-root' directory
+# You can use 'tree' command to view the directory content
+tree squashfs-root
+```
+## Run The Snap
+
+To run a shell in the confined environment:
+
+```shell
+$ sudo snap run --shell deviceupdate-agent
+```
+
+## View Snap Logs using Journalctl
+
+The name of the Device Update agent in the confined environment is `snap.deviceupdate-agent.deviceupdate-agent`.
+
+To view the journal log, use following command:
+
+```shell
+$ journalctl -u snap.deviceupdate-agent.deviceupdate-agent -f --no-tail
+```
+
+## Start and Stop The Device Update Snap Daemon
+
+To start the daemon:
+
+```shell
+$ sudo systemctl start  snap.deviceupdate-agent.deviceupdate-agent
+```
+
+To stop the daemon:
+
+```shell
+$ sudo systemctl stop  snap.deviceupdate-agent.deviceupdate-agent
+```
+
+To restart the daemon:
+
+```shell
+$ sudo systemctl restart  snap.deviceupdate-agent.deviceupdate-agent
+```
+
+## Provisioning Device Update Agent Snap
+
+> NOTE | To obtain comprehensive information on provisioning the Device Update Agent, please refer to the following link: [Device Update Agent Provisioning](https://learn.microsoft.com/en-us/azure/iot-hub-device-update/device-update-agent-provisioning)
+
+To configure the DU Agent, you can use `deviceupdate-agent.set-config` command.
+
+  This command decodes and saves base64 encoded configuration data to a file with the given name in the `\$SNAP_DATA/config` directory.  
+
+  The script takes two required options - the name of the config file and the base64 encoded data - and saves the data to the specified file.  
+
+  > NOTE | Only two config file names are allowed: `du-config.json` and `du-diagnostics-config.json`.
+
+| Options | Arguments | Description |
+|---|---|---|
+|-c, --config-file | \<config file name\> |  Specifies the name of the configuration file to be created or updated.<br/><br/>The only valid values for the config file name are \"du-config.json\" and \"du-diagnostics-config.json\".|
+| -d, --data  | \<base64 encoded data\> | Specifies the base64 encoded configuration data to be saved to the specified config file. |
+| -h, --help  | | Displays this usage information |
+
+Examples:
+
+```shell
+# Save the base64 encoded configuration data to the \"du-config.json\" file:
+
+$ sudo snap $COMMAND_NAME --config-file du-config.json --data "$(base64 /path/to/du-config.json)"
+```
+
+```shell
+# Save the base64 encoded configuration data to the "du-diagnostics-config.json" file:
+
+$ sudo snap $COMMAND_NAME -c du-diagnostics-config.json -d $(base64 /path/to/du-diagnostics-config.json)"
```

Comment:
should the `$(...)` be inside double-quotes like on line 390 for consistency?

Reply:
Fixed. thx for the suggestion.